int32 QDC_SIG1
int32 QDC_SIG2
int32 QDC_SIG3
int32 QDC_SIG4
int32 QDC_SIG5
int32 QDC_SIG6
int32 QDC_SIG7
int32 tool_select
